api/soap/mc_api.php

```diff
@@ -139,11 +139,22 @@ function mc_version() {
  */
 function mc_login( $p_username, $p_password ) {
 	$t_user_id = mci_check_login( $p_username, $p_password );
-	if( $t_user_id === false ) {
-		return mci_soap_fault_login_failed();
+	if( false === $t_user_id ) {
+		$t_user_id = mci_check_login( $p_username, $p_password, API_TOKEN_REPORT_ONLY );
```

Comment:
Re-work this to work with scopes instead of token types.  Expect a proper user name that is associated with the token.